### PR TITLE
Update work pools doc for flow and clarity

### PR DESCRIPTION
**Pull request by desertaxle**
This PR updates the work pools doc for flow and clarity. Most of the removed sections are housed in other places in the docs. The one exception is information about `flow.deploy`, which doesn't have a home right now. My plan is to create a separate page for python-based deployments in a subsequent PR.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 1]
```diff
@@ -156,106 +184,136 @@ If you don't use the `--type` flag to specify an infrastructure type, you are pr
...
+For example:
+
+```bash
+                               Work pools
+┏━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┓
+┃ Name       ┃    Type        ┃                                   ID ┃ Concurrency Limit ┃
+┡━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━┩
+│ barbeque   │ docker         │ 72c0a101-b3e2-4448-b5f8-a8c5184abd17 │ None              │
+│ k8s-pool   │ kubernetes     │ 7b6e3523-d35b-4882-84a7-7a107325bb3f │ None              │
+| my-pool    │ process        │ cd6ff9e8-bfd8-43be-9be3-69375f7a11cd │ None              │
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:213]:
```suggestion
 │ my-pool    │ process        │ cd6ff9e8-bfd8-43be-9be3-69375f7a11cd │ None              │
```

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 2]
```diff
@@ -156,106 +184,136 @@ If you don't use the `--type` flag to specify an infrastructure type, you are pr
...
-(for example, type, minimum, maximum).
+### Manage concurrency
 
-The job configuration section takes the values provided in the variables section and translates them into a configuration. 
-It then gives this configuration to a worker when it executes a flow run.
+Each work pool can optionally restrict concurrent runs of matching flows.
 
-The values in the `job_configuration` can use placeholders to reference values provided in the `variables` section. 
-Declare placeholders with double curly braces, for example, `{{ variable_name }}`. You can also hard-code `job_configuration` values.
+For example, a work pool with a concurrency limit of 5 only releases new work if less than five matching runs are in a
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:287]:
```suggestion
For example, a work pool with a concurrency limit of 5 only releases new work if fewer than five matching runs are in a 
```

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 3]
```diff
@@ -156,106 +184,136 @@ If you don't use the `--type` flag to specify an infrastructure type, you are pr
...
+- `clear-concurrency-limit` clears any concurrency limits from a work pool
 
-```bash
-prefect work-pool create --type process above-ground
-```
+#### Base job template
 
-You see these configuration options available in the Prefect UI:
-![process work pool configuration options](/3.0rc/img/ui/process-work-pool-config.png)
+Each work pool has a base job template that controls the of the behavior of the worker executing flow runs from the work pool.
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:298]:
```suggestion
Each work pool has a base job template that controls the behavior of the worker executing flow runs from the work pool.
```

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 4]
```diff
@@ -156,106 +184,136 @@ If you don't use the `--type` flag to specify an infrastructure type, you are pr
...
+The base job template acts as a contract defining the configuration passed to the worker for each flow run and the options available 
+to deployment creators to customize worker behavior per deployment.
 
-You can see an example of a JSON-formatted base job template with the 'Advanced' tab.
+A base job template comprises a `job_configuration` section and a `variables` section.
 
-You can examine the default base job template for a given worker type by running:
+The `variables` section defines the fields available that can be customized per deployment or flow run. The `variables` section follows the 
+[OpenAPI specification](https://swagger.io/specification/), which allows work pool creators to place limits on provided values 
+(for example, type, minimum, maximum).
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:307]:
```suggestion
(for example, type, minimum, and maximum).
```

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 5]
```diff
@@ -311,6 +369,23 @@ prefect work-pool get-default-base-job-template --type process
 }
 ```
 
+Each work pool type is configured with a default base job template, which is a good place to get started.
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:372]:
```suggestion
Each work pool type is configured with a default base job template, which is a good place to make initial modifications. 
```

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 6]
```diff
@@ -311,6 +369,23 @@ prefect work-pool get-default-base-job-template --type process
 }
 ```
 
+Each work pool type is configured with a default base job template, which is a good place to get started. 
+The default base template defines values that pass to every flow run, but you can override on a per-deployment or per-flow run basis.
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:373]:
```suggestion
The default base template defines values that pass to every flow run, but you can override them on a per-deployment or per-flow run basis.
```

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 7]
```diff
@@ -311,6 +369,23 @@ prefect work-pool get-default-base-job-template --type process
...
+
+For example, you see these configuration options available in the Prefect UI for `process` work pool:
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:375]:
```suggestion
For example, these configuration options are available in the Prefect UI for `process` work pool:
```

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 8]
```diff
@@ -311,6 +369,23 @@ prefect work-pool get-default-base-job-template --type process
...
+![process work pool configuration options](/3.0rc/img/ui/process-work-pool-config.png)
+
+The default base job template allow you to:
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:378]:
```suggestion
The default base job template allows you to:
```

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 9]
```diff
@@ -347,130 +422,24 @@ Learn more about [overriding job variables](/3.0rc/deploy/infrastructure-concept
...
-
-To resume a work pool through the Prefect CLI, use the `prefect work-pool resume` command with the work pool name.
-
-To delete a work pool through the Prefect CLI, use the `prefect work-pool delete` command with the work pool name.
-
-### Manage concurrency
+### Work queues
 
-Each work pool can optionally restrict concurrent runs of matching flows.
+Work queues offer advanced control over how runs are executed. Each work pool has a "default" queue which is used if a specific work queue name is not specified.
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:427]:
```suggestion
Work queues offer advanced control over how runs are executed. Each work pool has a "default" queue which is used if another  work queue name is not specified. 
```

**Review by discdiver — COMMENTED** [review]
Thank you! Minor edits.

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 10]
```diff
@@ -156,106 +184,136 @@ If you don't use the `--type` flag to specify an infrastructure type, you are pr
...
+To pause a work pool through the Prefect CLI, use the `prefect work-pool pause` command:
 
-Each work pool has a base job template that allows the customization of the behavior of the worker executing flow runs from the work pool.
+```bash
+prefect work-pool pause 'test-pool'
+```
 
-The base job template acts as a contract defining the configuration passed to the worker for each flow run and the options available 
-to deployment creators to customize worker behavior per deployment.
+To resume a work pool through the Prefect CLI, use the `prefect work-pool resume` command with the work pool name.
```
Comment (cicdw) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:279]: might be worth a `<Note>` in this section to mention that pausing work pools does _not_ pause underlying deployments which may continue to get scheduled

**Review by cicdw — COMMENTED** [review]
very nice - one small suggestion to add a `<Note>`, otherwise LGTM

File: docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx [hunk 11]
```diff
@@ -210,7 +210,7 @@ For example:
 ┡━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━┩
 │ barbeque   │ docker         │ 72c0a101-b3e2-4448-b5f8-a8c5184abd17 │ None              │
 │ k8s-pool   │ kubernetes     │ 7b6e3523-d35b-4882-84a7-7a107325bb3f │ None              │
-| my-pool    │ process        │ cd6ff9e8-bfd8-43be-9be3-69375f7a11cd │ None              │
+ │ my-pool    │ process        │ cd6ff9e8-bfd8-43be-9be3-69375f7a11cd │ None              │
```
Comment (cicdw) [inline docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:213]: @discdiver just curious - why was this change needed?

Comment (discdiver) [reply docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:213]: It looked like that pipe character was different than all the others (my syntax highlighter had it in different color), so I just copied one of the other pipes for consistency.

Comment (cicdw) [reply docs/3.0rc/deploy/infrastructure-concepts/work-pools.mdx:213]: ah i see what you mean, thanks
